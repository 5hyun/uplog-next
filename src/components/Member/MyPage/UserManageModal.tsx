import React, { useCallback, useEffect, useState } from 'react';
import {
  Flex,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
} from '@chakra-ui/react';
import { Link, useNavigate } from 'react-router-dom';
import { AiOutlineEye, AiOutlineEyeInvisible } from 'react-icons/ai';
import useInput from '@/hooks/useInput.ts';
type MessageType = 'success' | 'error' | 'warning';
import { useMutation } from 'react-query';
import { changePassword, deleteAccount } from '@/api/Members/mypage.ts';
import { user } from '@/recoil/User/atom.ts';
import { SetterOrUpdater, useRecoilState } from 'recoil';

interface Props {
  isOpen: boolean;
  onClose: () => void;
  isClickPwChange: boolean;
  setMessageInfo: SetterOrUpdater<{ type: MessageType; content: string } | null>;
}
export default function UserManageModal({
  isOpen,
  onClose,
  isClickPwChange,
  setMessageInfo,
}: Props) {
  const [userInfo, setUserInfo] = useRecoilState(user);
  const [password, onChangePassword, setPassword] = useInput('');
  const [newPassword, onChangeNewPassword, setNewPassword] = useInput('');
  const [isCheckPw, setIsCheckPw] = useState(false);
  const [isPwVisible, setIsPwVisible] = useState(false);
  const navigator = useNavigate();

  const resetPw = useCallback(() => {
    setPassword('');
    setNewPassword('');
    onClose();
  }, [onClose]);

  const { mutate: changePasswordMutate } = useMutation(changePassword, {
    onSuccess: (data) => {
      if ('message' in data) {
        setMessageInfo({ type: 'warning', content: data.message });
        return;
      }

      resetPw();
      setMessageInfo({ type: 'success', content: '비밀번호 변경 완료!' });
    },
    onError: () => {
      setMessageInfo({ type: 'error', content: '다시 시도해주세요.' });
    },
  });

  const { mutate: deleteAccountMutate } = useMutation(deleteAccount, {
    onSuccess: (data) => {
      if ('httpStatus' in data) {
        setMessageInfo({ type: 'warning', content: data.message });
        return;
      }

      sessionStorage.removeItem('userInfo');
      sessionStorage.removeItem('nowProduct');
      sessionStorage.removeItem('nowProject');
      sessionStorage.removeItem('nowTeamId');
      sessionStorage.removeItem('nowGroupId');
      setUserInfo(null);

      setMessageInfo({ type: 'success', content: '로그아웃 되었습니다.' });
      navigator('/');
    },
    onError: () => {
      setMessageInfo({ type: 'error', content: '다시 시도해주세요.' });
    },
  });

  const onClickChangePassword = useCallback(() => {
    if (!password || !newPassword) {
      setMessageInfo({ type: 'warning', content: '비밀번호를 입력해주세요.' });
      return;
    }

    if (!isCheckPw) {
      setMessageInfo({ type: 'warning', content: '올바른 새로운 비밀번호를 입력해주세요.' });
      return;
    }

    changePasswordMutate({ newPassword, password });
  }, [password, newPassword, userInfo, isCheckPw]);

  const onClickDeleteAccount = useCallback(() => {
    if (!password) {
      setMessageInfo({ type: 'warning', content: '비밀번호를 입력해주세요.' });
      return;
    }

    deleteAccountMutate({ password });
  }, [password, userInfo]);

  const onClickPwVisible = useCallback(() => setIsPwVisible((prev) => !prev), []);

  // 비밀번호 유효성 검사
  useEffect(() => {
    const regex = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]{8,15}$/;
    setIsCheckPw(regex.test(newPassword));
  }, [newPassword, isCheckPw]);

  return (
    <Modal isCentered onClose={resetPw} isOpen={isOpen}>
      <ModalOverlay />

      <ModalContent
        maxW="37.5rem"
        h={isClickPwChange ? '29rem' : '20rem'}
        shadow={'boxShadow-sign-up'}
        rounded={'none'}
        p={'1.2rem'}
        bg={'var(--white)'}
      >
        <ModalHeader
          fontSize={'1.8rem'}
          fontWeight={700}
          bg={'var(--white)'}
          color={'var(--black)'}
        >
          {isClickPwChange ? '비밀번호 변경' : '계정 삭제'}
        </ModalHeader>
        <ModalCloseButton
          fontSize={'1rem'}
          color={'var(--gray-light)'}
          mt={'0.6rem'}
          mr={'0.8rem'}
        />
        <ModalBody>
          <Flex justifyContent={'center'} h={'100%'}>
            <section className={'flex-col-center justify-evenly w-[24rem] h-full'}>
              {/* 현재 비밀번호 */}
              <div className={'w-full text-[1rem] mb-4'}>
                <span className={'text-gray-dark font-bold mb-[0.93rem]'}>현재 비밀번호</span>
                <input
                  type="password"
                  data-cy={'nowPasswordInput'}
                  value={password}
                  onChange={onChangePassword}
                  placeholder={'현재 비밀번호를 입력하세요.'}
                  maxLength={15}
                  className={
                    'w-full h-11 border-base border-gray-border text-[0.93rem] rounded-xl mb-2 p-4 text-black'
                  }
                />
                {isClickPwChange && (
                  <div className={'flex-row-center justify-start pl-4 text-sm text-gray-light'}>
                    <span>비밀번호를 잊으셨나요?</span>
                    <Link to={'/pwinquiry'} className={'ml-2 underline'}>
                      비밀번호 찾기
                    </Link>
                  </div>
                )}
              </div>
              {/* 새로운 비밀번호 */}
              {isClickPwChange && (
                <div className={'w-full text-[1rem]'}>
                  <span className={'text-gray-dark font-bold mb-[0.93rem]'}>새로운 비밀번호</span>
                  <span
                    className={
                      'flex-row-center border-base border-gray-border rounded-xl h-11 pl-4 mb-2 bg-border'
                    }
                  >
                    <input
                      type={`${isPwVisible ? 'text' : 'password'}`}
                      data-cy={'newPasswordInput'}
                      value={newPassword}
                      onChange={onChangeNewPassword}
                      placeholder={'새로운 비밀번호를 입력하세요.'}
                      maxLength={15}
                      className={'w-[90%] h-full text-[0.93rem] text-black'}
                    />
                    <span className={'w-[10%] text-2xl cursor-pointer'} onClick={onClickPwVisible}>
                      {isPwVisible ? (
                        <AiOutlineEyeInvisible className={'fill-gray-light'} />
                      ) : (
                        <AiOutlineEye className={'fill-gray-light'} />
                      )}
                    </span>
                  </span>
                  {!isCheckPw && newPassword && (
                    <span className={'flex-row-center justify-start text-sm text-[#E06469] pl-4'}>
                      영문/숫자/특수문자 포함, 8~15자로 입력해주세요.
                    </span>
                  )}
                </div>
              )}
            </section>
          </Flex>
        </ModalBody>

        <ModalFooter>
          <button
            className={`bg-orange rounded font-bold text-sm text-white
            ${isClickPwChange ? 'w-[4.5rem] h-9' : 'w-[6rem] h-9'}`}
            data-cy={'changePasswordButton'}
            onClick={() => {
              isClickPwChange ? onClickChangePassword() : onClickDeleteAccount();
            }}
          >
            {isClickPwChange ? '확인' : '계정 삭제'}
          </button>
        </ModalFooter>
      </ModalContent>
    </Modal>
  );
}
