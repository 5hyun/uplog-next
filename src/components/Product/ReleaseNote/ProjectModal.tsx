import React, { useCallback, useEffect, useRef } from 'react';
import {
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Tooltip,
} from '@chakra-ui/react';
import useInput from '@/hooks/useInput.ts';
import { BsQuestionCircle } from 'react-icons/bs';
import CompleteModalTooltip from '@/components/Member/MyPage/CompleteModalTooltip.tsx';
import { completeProject, createProject } from '@/api/Project/Version.ts';
import { useMutation, useQueryClient } from 'react-query';
import { ProductInfo } from '@/typings/product.ts';
import { eachProductProjects } from '@/recoil/Project/atom.ts';
import { useRecoilState } from 'recoil';
import { Release } from '@/typings/project.ts';
type MessageType = 'success' | 'error' | 'warning';

interface Props {
  isOpen: boolean;
  onClose: () => void;
  isAdd: boolean;
  versionName: string;
  showMessage: (type: MessageType, content: string) => void;
  nowProjectId: number;
}
export default function ProjectModal({
  isOpen,
  onClose,
  isAdd,
  versionName,
  showMessage,
  nowProjectId,
}: Props) {
  const nowProduct: ProductInfo = JSON.parse(sessionStorage.getItem('nowProduct')!);
  const queryClient = useQueryClient();

  // 프로젝트 추가에서는 프로젝트 이름, 완료에서는 최종 버전
  const [text, onChangeText, setText] = useInput('');
  const [projects, setProjects] = useRecoilState(eachProductProjects);
  const newProjectId = useRef(-1);

  const { mutate: createProjectMutate } = useMutation(createProject, {
    onSuccess: (data) => {
      if (typeof data !== 'string') {
        newProjectId.current = data.id;
      }
    },
    onMutate: async ({ version }) => {
      await queryClient.cancelQueries(['getAllProductProjects', nowProduct?.productId]);

      const snapshot = queryClient.getQueryData(['getAllProductProjects', nowProduct?.productId]);

      queryClient.setQueriesData(['getAllProductProjects', nowProduct?.productId], () => {
        const temp: Release[] = [...projects, { version, projectStatus: 'PROGRESS_IN', id: -1 }];

        return temp;
      });

      return { snapshot };
    },
    onError: (error, newTodo, context) => {
      queryClient.setQueriesData(
        ['getAllProductProjects', nowProduct?.productId],
        context?.snapshot
      );
    },
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey: ['getAllProductProjects', nowProduct?.productId] });
    },
  });

  const { mutate: completeProjectMutate } = useMutation(completeProject, {
    onMutate: async ({ version, projectId }) => {
      await queryClient.cancelQueries(['getAllProductProjects', nowProduct?.productId]);

      const snapshot = queryClient.getQueryData(['getAllProductProjects', nowProduct?.productId]);

      queryClient.setQueriesData(['getAllProductProjects', nowProduct?.productId], () => {
        const temp: Release[] = projects.map((project) => {
          if (project.id === projectId) {
            return { ...project, version, projectStatus: 'PROGRESS_COMPLETE' };
          } else return project;
        });

        setProjects(temp);
        return temp;
      });

      return { snapshot };
    },
    onError: (error, newTodo, context) => {
      queryClient.setQueriesData(
        ['getAllProductProjects', nowProduct?.productId],
        context?.snapshot
      );
    },
    onSettled: () => {
      queryClient.invalidateQueries({ queryKey: ['getAllProductProjects', nowProduct?.productId] });
    },
  });

  const onClickCreateProject = useCallback(() => {
    if (projects.some((project) => project.version === text)) {
      showMessage('error', '이미 존재하는 버전입니다!');
      return;
    }

    //TODO: url 수정 필요
    createProjectMutate({ productId: nowProduct?.productId, version: `${text}(임시)`, link: '/' });
    showMessage('success', '프로젝트가 생성되었습니다!');
    onClose();
    setText('');
  }, [text]);

  const onClickCompleteProject = useCallback(() => {
    if (projects.some((project) => project.version === text)) {
      showMessage('error', '이미 존재하는 버전입니다!');
      return;
    }

    completeProjectMutate({
      projectId: nowProjectId === -1 ? newProjectId.current : nowProjectId,
      version: text,
    });
    showMessage('success', '프로젝트 완료!');
    onClose();
    setText('');
  }, [text, newProjectId]);

  useEffect(() => {
    if (versionName && !isAdd) {
      setText(versionName.replace('(임시)', ''));
    } else setText('');
  }, [isAdd]);

  return (
    <Modal isCentered onClose={onClose} isOpen={isOpen}>
      <ModalOverlay />

      <ModalContent
        maxW={'41rem'}
        h={'22rem'}
        shadow={'boxShadow-sign-up'}
        rounded={'none'}
        p={'1.2rem'}
        bg={'var(--white)'}
      >
        <ModalHeader
          fontSize={'1.8rem'}
          fontWeight={700}
          bg={'var(--white)'}
          color={'var(--black)'}
        >
          {isAdd ? '프로젝트 추가' : '프로젝트 완료'}
        </ModalHeader>
        <ModalCloseButton
          fontSize={'1rem'}
          color={'var(--gray-light)'}
          mt={'0.6rem'}
          mr={'0.8rem'}
        />
        <ModalBody>
          <section className={'flex flex-col justify-center items-start w-h-full'}>
            <div className={'mx-auto'}>
              <span className={'text-gray-dark font-bold text-[1.2rem]'}>
                {isAdd ? '버전 입력' : '최종 버전 입력'}
              </span>
              <div className={'flex items-center'}>
                <input
                  className={`border-base mt-4 mb-2 ${
                    isAdd ? 'w-[29rem]' : 'w-[25rem]'
                  } h-14 rounded-xl px-4 py-2 text-[1.1rem] text-black `}
                  type="text"
                  value={text}
                  onChange={onChangeText}
                  placeholder={
                    isAdd ? '새로 생성할 버전을 입력해 주세요.' : '최종 버전을 입력해 주세요.'
                  }
                  maxLength={10}
                />

                {!isAdd && (
                  <Tooltip
                    placement={'bottom-start'}
                    label={<CompleteModalTooltip />}
                    color={'var(--black)'}
                    backgroundColor={'var(--white)'}
                    minW={'29rem'}
                    boxShadow={
                      '2px 2px 5px 1px rgba(0, 0, 0, 0.05), -2px -2px 5px 0px rgba(0, 0, 0, 0.05)'
                    }
                    borderRadius={'0.31rem'}
                  >
                    <span className={'flex-col-center mt-3 ml-4 text-gray-dark cursor-pointer '}>
                      <BsQuestionCircle className={'text-3xl mb-1'} />
                      버전
                    </span>
                  </Tooltip>
                )}
              </div>
              {isAdd && (
                <span className={'text-gray-light text-[0.6rem] ml-2'}>
                  * 현재 생성되는 버전은 임시 버전이며, 완료 시 최종 버전명을 재설정할 수 있습니다.
                </span>
              )}
            </div>
          </section>
        </ModalBody>

        <ModalFooter>
          <button
            className={'bg-orange rounded font-bold text-sm text-white w-[4.5rem] h-9'}
            onClick={() => {
              isAdd ? onClickCreateProject() : onClickCompleteProject();
            }}
          >
            완료
          </button>
        </ModalFooter>
      </ModalContent>
    </Modal>
  );
}
