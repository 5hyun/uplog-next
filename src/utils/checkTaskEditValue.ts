import { TaskData } from '@/typings/task.ts';

export function checkTaskEditValue(editTask: TaskData): boolean {
  if (editTask.taskName === '') {
    return false;
  }
  if (editTask.targetMemberInfoDTO && editTask.targetMemberInfoDTO.id === 0) {
    return false;
  }
  if (editTask.menuId === 0) {
    return false;
  }
  if (editTask.teamId === 0) {
    return false;
  }
  if (editTask.startTime === '' || editTask.endTime === '') {
    return false;
  }

  return true;
}
