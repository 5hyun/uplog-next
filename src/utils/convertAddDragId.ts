import { StatusTaskData, Tasks } from '@/typings/task.ts';
import { useState } from 'react';

export function convertAddDragId(tasks: StatusTaskData): Tasks {
  const [convertedTasks, setConvertedTasks] = useState<Tasks>({
    PROGRESS_BEFORE: [],
    PROGRESS_IN: [],
    PROGRESS_COMPLETE: [],
  });

  const beforeStatusTasks = tasks.PROGRESS_BEFORE.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_BEFORE: beforeStatusTasks,
  });

  const inStatusTasks = tasks.PROGRESS_IN.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_IN: inStatusTasks,
  });

  const completeStatusTasks = tasks.PROGRESS_COMPLETE.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_COMPLETE: completeStatusTasks,
  });

  return convertedTasks;
}
