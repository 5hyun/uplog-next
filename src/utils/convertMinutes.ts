// ** 타이머 분:초 구하는 함수 */
export function convertMinutes(seconds: number): string {
  const minutes = Math.floor(seconds / 60);
  const remainingSeconds = seconds % 60;
  const formattedSeconds = remainingSeconds < 10 ? `0${remainingSeconds}` : `${remainingSeconds}`;

  if (minutes === 0) {
    return `00:${formattedSeconds}`;
  }

  return `${minutes}:${formattedSeconds}`;
}
