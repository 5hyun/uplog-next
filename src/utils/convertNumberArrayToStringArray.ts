export function convertNumberArrayToStringArray(numbers: number[]): string[] {
  // 배열의 각 요소를 문자열로 변환하여 새로운 배열을 생성합니다.
  const strings: string[] = numbers.map((number) => number.toString());
  return strings;
}
