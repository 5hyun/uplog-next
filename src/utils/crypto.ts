import CryptoJS from 'crypto-js';

/* Salt 비밀 키 */
// const salt = import.meta.env.VITE_SALT;
const salt = '1234';

/* 암호화 */
export const encrypt = (text: string) => {
  // 값이 없을 경우 빈 문자열 반환
  if (!text) return '';
  return CryptoJS.AES.encrypt(text, salt).toString();
};

/* 복호화 */
export const decrypt = (text: string) => {
  // 값이 없을 경우 빈 문자열 반환
  if (!text) return '';

  try {
    const bytes = CryptoJS.AES.decrypt(text, salt); // 복호화 시도
    return bytes.toString(CryptoJS.enc.Utf8);
  } catch (error) {
    return ''; // 에러 발생 시 빈 문자열 반환
  }
};
