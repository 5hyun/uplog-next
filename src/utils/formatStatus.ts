// ** 진행 전,중,후 상태로 변환하기 위한 함수 */

export function formatStatus(status: string): string {
  if (status === 'PROGRESS_BEFORE') {
    return '진행 전';
  } else if (status === 'PROGRESS_IN') {
    return '진행 중';
  }

  return '진행 후';
}
