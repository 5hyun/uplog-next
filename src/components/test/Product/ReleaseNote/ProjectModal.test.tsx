import { describe, expect, test } from '@jest/globals';
import { fireEvent, getByText, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { withRouter } from '@/tests/utils';
import { Route } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import UserManageModal from '@/components/Member/MyPage/UserManageModal';
import userEvent from '@testing-library/user-event';
import ProjectModal from '@/components/Product/ReleaseNote/ProjectModal';

describe('ProjectModal', () => {
  const props_isAdd = {
    isOpen: true,
    onClose: jest.fn(),
    isAdd: true,
    versionName: 'v.1.0.0',
  };

  const props_isComplete = {
    isOpen: true,
    onClose: jest.fn(),
    isAdd: false,
    versionName: 'v.1.0.0',
  };

  // 테스트가 끝날때 마다 Mock 함수를 깨끗하게 정리
  afterEach(() => props_isAdd.onClose.mockReset());

  // 프로젝트 추가하기
  test('isAdd modal correctly', () => {
    const component = renderer.create(
      withRouter(
        <Route path={'/workspace/product'} element={<ProjectModal {...props_isAdd} />} />,
        '/workspace/product'
      )
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('isAdd modal, open', () => {
    render(
      withRouter(
        <Route path={'/workspace/product'} element={<ProjectModal {...props_isAdd} />} />,
        '/workspace/product'
      )
    );

    expect(screen.getByText('프로젝트 추가')).toBeInTheDocument();
    expect(screen.getByText('버전 입력')).toBeInTheDocument();
    expect(screen.queryByText('프로젝트 완료')).not.toBeInTheDocument();
    expect(screen.queryByText('최종 버전 입력')).not.toBeInTheDocument();
  });

  test('isAdd modal, input', () => {
    render(
      withRouter(
        <Route path={'/workspace/product'} element={<ProjectModal {...props_isAdd} />} />,
        '/workspace/product'
      )
    );

    const versionInput = screen.getByPlaceholderText('새로 생성할 버전을 입력해 주세요.');
    expect(versionInput).toHaveAttribute('value', ''); // Replace the second parameter with the expected initial value

    fireEvent.change(versionInput, { target: { value: 'v.1.0.1' } });
    expect(versionInput).toHaveAttribute('value', 'v.1.0.1'); // Replace the second parameter with the expected initial value
  });

  // 프로젝트 완료하기
  test('isComplete modal correctly', () => {
    const component = renderer.create(
      withRouter(
        <Route path={'/workspace/product'} element={<ProjectModal {...props_isComplete} />} />,
        '/workspace/product'
      )
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('isComplete modal, open', () => {
    render(
      withRouter(
        <Route path={'/workspace/product'} element={<ProjectModal {...props_isComplete} />} />,
        '/workspace/product'
      )
    );

    expect(screen.queryByText('프로젝트 추가')).not.toBeInTheDocument();
    expect(screen.queryByText('버전 입력')).not.toBeInTheDocument();
    expect(screen.getByText('프로젝트 완료')).toBeInTheDocument();
    expect(screen.getByText('최종 버전 입력')).toBeInTheDocument();
  });

  test('isComplete modal, input', () => {
    render(
      withRouter(
        <Route path={'/workspace/product'} element={<ProjectModal {...props_isComplete} />} />,
        '/workspace/product'
      )
    );

    const versionInput = screen.getByPlaceholderText('최종 버전을 입력해 주세요.');
    expect(versionInput).toHaveAttribute('value', props_isComplete.versionName); // Replace the second parameter with the expected initial value

    fireEvent.change(versionInput, { target: { value: 'v.1.0.2' } });
    expect(versionInput).toHaveAttribute('value', 'v.1.0.2'); // Replace the second parameter with the expected initial value
  });
});
