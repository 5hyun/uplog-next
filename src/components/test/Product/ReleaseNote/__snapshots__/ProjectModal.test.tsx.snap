// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectModal isAdd modal correctly 1`] = `<span />`;

exports[`ProjectModal isComplete modal correctly 1`] = `<span />`;
